File: algorithm_python/Algorithm_Session/Day1/boj_1110.py
n = int(input())

cnt = 1
rst = n

while True:
    x1 = rst // 10
    x2 = rst % 10
    y1 = (x1 + x2) // 10
    y2 = (x1 + x2) % 10

    rst = x2 * 10 + y2
    if rst == n:
        break
    cnt += 1

print(cnt)

File: 02_Algorithm/algorithm_java/Week1/OT/boj_1110.java
package Week1.OT;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj_1110 {
    public static void main(String[] args) throws IOException {
        var s = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(s.readLine());
        int rst = n;
        int cnt = 1;

        while (true) {
            int a = rst % 10;
            int b = ((rst / 10) + (rst % 10)) % 10;
            rst = (a * 10) + b;

            if (rst == n) break;
            cnt++;
        }

        System.out.print(cnt);
        s.close();
    }
}
